### DISCUSSION TITLE
Python sidecar container pip install hashin fails

### DISCUSSION THREAD

**Discussion by vbode** [renovatebot/renovate#9534] [Request Help]
Hello,

I am facing a similar issue to this [one](https://github.com/renovatebot/renovate/discussions/8169). The error is slightly different though:

`"message": "Command failed: docker run --rm --name=renovate-chunkpython --label=renovate-chunkchild --user=1003:0 -v \"/opt/renovatebot/repos/gitlab/VincentB/authentication-services\":\"/opt/renovatebot/repos/gitlab/VincentB/authentication-services\" -v \"/opt/renovatebot/\":\"/opt/renovatebot/\" -w \"/opt/renovatebot/repos/gitlab/VincentB/authentication-services\" docker.io/renovate/python bash -l -c \"pip install hashin\"\nWARNING: The directory '/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.\nERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: '/.local'\nCheck the permissions.\n\n",`

I have tried setting the cacheDir setting but that did not help. The main renovate container is running with this configuration:
docker run --rm --pull always\
            -e "GITHUB_COM_TOKEN={{ github_com_token }}" \
            -e "LOG_LEVEL=$2" \
            -e "HOME=/tmp" \
            -e "NPM_CONFIG_PREFIX=/tmp/.npm" \
            -v "/tmp:/tmp" \
            -v "/var/run/docker.sock:/var/run/docker.sock" \
            -v "$1:/usr/src/app/config.js" \
            -v "/opt/renovatebot/:/opt/renovatebot/" \
            -u 1003:999
            
Any ideas what could be the issue?

**Comment by viceice** [comment]
Why you change the user? If you need that, please make sure to use root group. Also set `dockerUser` to same settings.

**Reply by viceice** [reply]
Also use slim image for docker binary source

**Reply by vbode** [reply]
Hi,
I tried using the root group for the docker run command (-u 1003:0) but that leads to:
`Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=renovate%2Fpython&tag=3.9.3: dial unix /var/run/docker.sock: connect: permission denied`

In the current setup dockerUser is set to 1003:0 which works, except for the error above. I have also tested with the slim image.

Any other suggestions ? :)

**Reply by viceice** [reply]
Why you are using user 1003 and not our prepared user 1000 (ubuntu) ?

If you are using the full image, you don't need mount the docker socket.

Can you describe in more detail what do you like to achive?

**Reply by vbode** [reply]
We run renovate in this way:
- From a gitlab repository, there is a schedule to run renovate at night for a list of repositories.
- The schedule kicks off a gitlab ci pipeline. The gitlab runner then executes a renovate script (with the above docker run command) on a different machine via SSH using a user called renovatebot (1003). The renovatebot user has permissions for /opt/renovatebot.
- The sidecar container is needed because some repositories are using npm v7.

So far this setup is working, except for this permissions issue now for `/.cache/pip`

**Reply by viceice** [reply]
You should use the slim image then, saves a lot download size of renovate image.

Can you please test using default  renovate docker users? it should work.

For what do you use `/opt/renovatebot` ? can should share your sanitized `config.js` ?

**Reply by vbode** [reply]
I will adjust to use the slim image.

`/opt/renovatebot` was set as the baseDir (and also as the path for the logFile. I have removed this (and removed the docker user in the run command and config), and now the error does not occur anymore.

Thanks for the help!
